### PR TITLE
BREAK: drop old python versions, refactor CI

### PR DESCRIPTION
**Pull request by aikikode**
NB: do not merge until https://github.com/kpn-digital/py-fqn-decorators/pull/19 is merged and we have tag 2.0.0 for `fqn-decorators` to check `py-timeexecution` works with version >=2.0 

Changes here:
- drop `pkgversion`, Python 2.7 and 3.5 support
- replace `docker-compose` with `tox-docker`
- update `elasticsearch`, `zookeeper` and `kafka` docker images
- remove `fqn-decorators < 2` restriction (with `... < 3`)
- add `mypy` lint checks
- update `isort`, `flake8` and `black` tools configuration

### CODE REVIEW

**Review by igbun — APPROVED** [review]

**Review by kammala — CHANGES_REQUESTED** [review]

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+from typing import List
+
+# setuptools doesn't support type hints for now:
+# https://github.com/pypa/setuptools/issues/2345
+# so we ignoring mypy checks on this package
+from setuptools import find_packages, setup
+
+
+def list_requirements(path: str) -> List[str]:
```
Comment (kammala) [inline setup.py:11]: I would rather move these files content in here and used `.[influxdb,kafka,elasticsearch]` somewhere in tox

File: requirements/requirements-base.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 pkgsettings>=0.9.2
 six>=1.10.0,<2.0.0
```
Comment (kammala) [inline requirements/requirements-base.txt:2]: do we need six as we are dropping `py27`?

File: setup.cfg [hunk 3]
```diff
@@ -1,22 +1,17 @@
...
+ignore =
+    # flake8 misbehaves: https://github.com/PyCQA/pycodestyle/issues/373
+    E203
+    # duplicate for B950
+    E501
+    E902
+    # mutual exclusive with W504
+    W503
+    # conflicts with FastAPI `Query`, `Header`, `Body` functions
+    B008
```
Comment (kammala) [inline setup.cfg:14]: might not need that :thinking:

**Review by kammala — CHANGES_REQUESTED** [review]

File: setup.cfg [hunk 4]
```diff
@@ -1,22 +1,17 @@
...
-from_first = false
-length_sort = false
-order_by_type = true
-use_parenthesis = true
-line_length = 120
-force_grid_wrap = 0
-
-[wheel]
-universal = 1
+[mypy-pip.*,influxdb.*,elasticsearch.*,fqn_decorators.*,pkgsettings.*,kafka.*,setuptools.*,Queue.*]
```
[the remark this answers is not in the document]
Comment (aikikode) [reply setup.cfg]: It means to skip mypy checks for these libraries as they don't support type checks yet.

File: tests/test_influxdb.py [hunk 5]
```diff
@@ -1,17 +1,22 @@
...
+
 from fqn_decorators import get_fqn
 from influxdb.influxdb08.client import InfluxDBClientError
+
 from tests.conftest import Dummy, go
 from tests.test_base_backend import TestBaseBackend
 from time_execution import settings
 from time_execution.backends.influxdb import InfluxBackend
 
+INFLUX_HOST = os.getenv("INFLUX_HOST")
```
Comment (igbun) [inline tests/test_influxdb.py:11]: Perhaps, not relevant for this PR. But we should maybe also drop InfluxDB support.

**Review by igbun — DISMISSED** [review]

File: tests/test_influxdb.py (hunk 5 above)
Comment (kammala) [reply tests/test_influxdb.py:11]: but there are so many people using it all over the world with `timeexecution` ;)

Comment (kammala) [reply tests/test_influxdb.py:11]: what about kafka? it was never tested on prod. should we drop it as well?

**Review by kammala — APPROVED** [review]

**Comment by kammala** [comment]
LGTM, but coverage has decreased for some reason.

**Pull request merged**